### ISSUE TITLE
Upgrade 1.7.0beta4 -> 1.7.0

### ISSUE THREAD

**Issue by rrondeau** [hashicorp/consul#7269]
#### Overview of the Issue

I am trying to upgrade my cluster from beta4 to the stable release.
I am using the consul-hem chart.

#### Reproduction Steps

Upgrade an existing 1.7.0-beta4 server to 1.7.0

### Operating system and Environment details

Inside GKE cluster with consul-helm chart depoyment

### Log Fragments

```
    2020-02-12T09:02:40.738Z [DEBUG] agent.tlsutil: Update: version=1
    2020-02-12T09:02:40.754Z [DEBUG] agent.tlsutil: OutgoingRPCWrapper: version=1
    2020-02-12T09:02:40.913Z [ERROR] agent.server.raft: failed to restore snapshot: id=950-20761235-1581497596260 error="codec.decoder: Only encoded map or array can be decoded into a struct. (valueType: 6)"
    2020-02-12T09:02:40.988Z [ERROR] agent.server.raft: failed to restore snapshot: id=950-20743636-1581497008238 error="codec.decoder: Only encoded map or array can be decoded into a struct. (valueType: 6)"
    2020-02-12T09:02:40.988Z [INFO]  agent.server: shutting down server
    2020-02-12T09:02:40.989Z [ERROR] agent: Error starting agent: error="Failed to start Consul server: Failed to start Raft: failed to load any existing snapshots"
    2020-02-12T09:02:40.989Z [INFO]  agent: Exit code: code=1
```

**Comment by mkeeler** [comment]
@rrondeau Unfortunately this is expected behavior when updating from beta4 to the final release. This information should have been in the release announcement and I apologize for it being missing.

During the beta period we discovered that there were issues just like you are seeing when upgrading from 1.6.3 to 1.7.0-beta4 (#7223). After looking into it we found that we had inadvertently introduced some incompatibilities into a couple data structures during the beta period. We fixed those incompatibilities so that once the final version was released users production systems running 1.6.x (or other releases) could upgrade without issue. The downside to this approach was that it would make upgrades from beta4 to the final release not possible.

Unfortunately there is no solution other than to remove all the existing data in the cluster.
